cron/src/main/scala/com/criteo/cuttle/cron/CronScheduler.scala

```diff
@@ -150,8 +176,15 @@ case class CronScheduler(logger: Logger) extends Scheduler[CronScheduling] {
             for {
               _ <- IO(state.addNextEventToState(job, scheduledAt.instant))
               _ <- logState
-              _ <- Timeout.applyF(scheduledAt.delay)
-              _ <- runAndRetry(job, scheduledAt, 0)
+              deferred <- Deferred[IO, Unit]
+              _ <- IO(state.addDeferred(job, deferred))
+              scheduledAt <- IO.race(
+                //Sleep and return scheduledAt
+                IO.sleep(scheduledAt.delay).map(_ => scheduledAt),
+                //Woken by user runNow operation, return new time.
+                deferred.get.map(_ => ScheduledAt(Instant.now, FiniteDuration(0, TimeUnit.SECONDS))))
+              _ <- IO(state.removeDeferred(job, deferred))
+              _ <- runAndRetry(job, scheduledAt.merge, 0)
```

Comment:
Sorry I didn't get what does `.merge` do?

Reply:
merge gets the one of the Either that is filled.